common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -746,6 +746,8 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         "This value must not contain any special character used in HDFS URI (e.g., ':', '%', '/' etc). \n" +
         "The user has to be aware that the dynamic partition value should not contain this value to avoid confusions."),
     DEFAULT_ZOOKEEPER_PARTITION_NAME("hive.lockmgr.zookeeper.default.partition.name", "__HIVE_DEFAULT_ZOOKEEPER_PARTITION__", ""),
+    HIVE_EXEC_SCHEDULED_POOL_NUM_THREADS("hive.exec.scheduler.num.threads", 10,
```

Comment:
Yes, the role of corepoolsize is exactly what you understand.

The reason I set to 10 is just because `metastore.thread.pool.size` is 10. I think 2 would be good.

Reply:
https://github.com/EugeneChung/hive/commit/ccf6ed68b62cfe71a9014abd4e58cb7eeac44d44 I set it to 2.